### ISSUE TITLE
ZODB persistence problems with annotations within a test (as part of a plone.rest testsuite)

### ISSUE THREAD

**Issue by zopyx** [plone/Products.CMFPlone#1378]
I have a ZODB persistence issue with a test of XML Director where a plone.rest API method modifies an annotations and metadata of an object. The change in a Plone 4.3 setup is persisted but no with Plone 5.

The related code of the plone.rest API method is here:

https://github.com/xml-director/xmldirector.plonecore/blob/cleanup/xmldirector/plonecore/browser/restapi.py#L258-L284

```
bin/test . test_set_get
/home/ajung/.buildout/eggs/setuptools-19.2-py2.7.egg/pkg_resources/__init__.py:213: RuntimeWarning: You have iterated over the result of pkg_resources.parse_version. This is a legacy behavior which is inconsistent with the new version class introduced in setuptools 8.0. In most cases, conversion to a tuple is unnecessary. For comparison of versions, sort the Version instances directly. If you have another use case requiring the tuple, please file a bug with the setuptools project describing that need.
Running xmldirector.plonecore.tests.apibase.API:Functional tests:
  Set up plone.testing.zca.LayerCleanup in 0.000 seconds.
  Set up plone.testing.z2.Startup in 0.174 seconds.
  Set up plone.app.testing.layers.PloneFixture in 6.922 seconds.
  Set up xmldirector.plonecore.tests.apibase.APILayer/home/ajung/.buildout/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/fields.py:417: UserWarning: You did not specify an i18n translation domain for the 'description' field in /home/ajung/.buildout/eggs/zopyx.plone.persistentlogger-0.2.5-py2.7.egg/zopyx/plone/persistentlogger/configure.zcml
  "'%s' field in %s" % (self.getName(), context.info.file )
/home/ajung/.buildout/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/fields.py:417: UserWarning: You did not specify an i18n translation domain for the 'title' field in /home/ajung/.buildout/eggs/zopyx.plone.persistentlogger-0.2.5-py2.7.egg/zopyx/plone/persistentlogger/configure.zcml
  "'%s' field in %s" % (self.getName(), context.info.file )
 in 0.558 seconds.
  Set up plone.testing.z2.ZServer in 0.501 seconds.
  Set up xmldirector.plonecore.tests.apibase.API:Functional in 0.000 seconds.
  Running:
    1/1 (100.0%)

Failure in test test_set_get_metadata (xmldirector.plonecore.tests.test_api.TestAPI)
Traceback (most recent call last):
  File "/opt/buildout.python/parts/opt/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/home/ajung/sandboxes/xmldirector.plonecore/xmldirector/plonecore/tests/test_api.py", line 118, in test_set_get_metadata
    self.assertEqual(metadata['title'], data['title'])
  File "/opt/buildout.python/parts/opt/lib/python2.7/unittest/case.py", line 513, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/opt/buildout.python/parts/opt/lib/python2.7/unittest/case.py", line 924, in assertMultiLineEqual
    self.fail(self._formatMessage(msg, standardMsg))
  File "/opt/buildout.python/parts/opt/lib/python2.7/unittest/case.py", line 410, in fail
    raise self.failureException(msg)
AssertionError: u'' != u'hello world'
+ hello world

The following test left new threads behind:
test_set_get_metadata (xmldirector.plonecore.tests.test_api.TestAPI)
New thread(s): [<_DummyThread(Dummy-1, started daemon 139993452377856)>]

  Ran 1 tests with 1 failures and 0 errors in 0.113 seconds.
Tearing down left over layers:
  Tear down xmldirector.plonecore.tests.apibase.API:Functional in 0.000 seconds.
  Tear down xmldirector.plonecore.tests.apibase.APILayer in 0.025 seconds.
  Tear down plone.testing.z2.ZServer in 5.379 seconds.
  Tear down plone.app.testing.layers.PloneFixture in 0.044 seconds.
  Tear down plone.testing.z2.Startup in 0.004 seconds.
  Tear down plone.testing.zca.LayerCleanup in 0.001 seconds.
```

The title set using setTitle() and retrieved using Title() are not identical.
Not sure if this issue is related to the plone.rest API setup or some issue with plone.app.testing.

**Comment by hvelarde** [comment]
probably related with #1362

**Comment by jensens** [comment]
seems to be fixed with solved #1362
